exporter/fileexporter/file_exporter.go

```diff
@@ -129,7 +130,7 @@ func (e *fileExporter) Shutdown(context.Context) error {
 }
 
 func buildExportFunc(cfg *Config) func(e *fileExporter, buf []byte) error {
-	if cfg.FormatType == formatTypeProto {
+	if cfg.FormatType == formatTypeProto || cfg.Compression != defaultCompression {
```

Comment:
I am unclear about the relationship between `format` and `compression` here. 

For example, why does `json && zstd` behave the same as `proto && none`?

Reply:
It's really not easy for other to understand here, I'll fix it